src/Workspaces/Remote/ServiceHub/UserOperationBooster.cs

```diff
@@ -10,60 +10,39 @@ namespace Microsoft.CodeAnalysis.Remote
     /// <summary>
     /// Boost performance of any servicehub service which is invoked by user explicit actions
     /// </summary>
-    internal static class UserOperationBooster
+    internal struct UserOperationBooster : IDisposable
     {
         private static int s_count = 0;
+        private static readonly object s_gate = new object();
 
-        public static IDisposable Boost()
+        public static UserOperationBooster Boost()
         {
-            return new Booster();
-        }
-
-        private static void Start()
-        {
-            var value = Interlocked.Increment(ref s_count);
```

Comment:
For example, we could have the following:

BG code is running (refcount is 1).
BG code ends, drops ref count to 0.
User code runs, increases refcount to 1.
User code sets priority to normal
BG code now runs and sets priority to low.